### ISSUE TITLE
无法运行Client

### ISSUE THREAD

**Issue by Huyuchao-x** [Murlors/Face-Recognition-Access-Control#1]
你好，请问以下配置完环境后能够运行Server，但是运行Client的时候报错请问应该怎么解决，错误如下：
Traceback (most recent call last):   File "D:/IdeaProject/Face-Recognition-Access-Control-main/Client/main.py", line 77, in <module>     gate.run()   File "D:\IdeaProject\Face-Recognition-Access-Control-main\Client\Gate.py", line 70, in run     small_frame = cv2.resize(rgb_frame, (0, 0), fx=0.25, fy=0.25) TypeError: Expected Ptr<cv::UMat> for argument 'src' [ WARN:0] global C:\Users\appveyor\AppData\Local\Temp\1\pip-req-build-b460jg4o\opencv\modules\videoio\src\cap_msmf.cpp (436) `anonymous-namespace'::SourceReaderCB::~SourceReaderCB terminating async callback

**Comment by GXshuyang** [comment]
It was addressed in <https://github.com/Murlors/Face-Recognition-Access-Control/pull/2#issue-1683324767> , specifically in the commit <https://github.com/GXshuyang/Face-Recognition-Access-Control/commit/620c3956cfe9b0bcfd4153c298fbea2d59c43cb1>

**Comment by Huyuchao-x** [comment]
> It was addressed in [#2 (comment)](https://github.com/Murlors/Face-Recognition-Access-Control/pull/2#issue-1683324767) , specifically in the commit [GXshuyang@620c395](https://github.com/GXshuyang/Face-Recognition-Access-Control/commit/620c3956cfe9b0bcfd4153c298fbea2d59c43cb1)
Solved, thank you. But I have another problem: my computer has integrated graphics and I installed pytrch -cpu, but at runtime it happens that：
 File "D:/IdeaProject/Face-Recognition-Access-Control-main/Server/ImgServer.py", line 25, in recognize
    rgb_frame = torch.tensor(rgb_frame).cuda().float()
  File "D:\ProgramData\Anaconda3\envs\Face-Recognition-Access-Control-main\lib\site-packages\torch\cuda\__init__.py", line 186, in _lazy_init
    _check_driver()
  File "D:\ProgramData\Anaconda3\envs\Face-Recognition-Access-Control-main\lib\site-packages\torch\cuda\__init__.py", line 61, in _check_driver
    raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled

How should I rewrite my program？

**Comment by Murlors** [comment]
> > > It was addressed in [#2 (comment)](https://github.com/Murlors/Face-Recognition-Access-Control/pull/2#issue-1683324767) , specifically in the commit [GXshuyang@620c395](https://github.com/GXshuyang/Face-Recognition-Access-Control/commit/620c3956cfe9b0bcfd4153c298fbea2d59c43cb1)
> > > 
> > Solved, thank you. But I have another problem: my computer has integrated graphics and I installed pytrch -cpu, but at runtime it happens that：
> > File "D:/IdeaProject/Face-Recognition-Access-Control-main/Server/ImgServer.py", line 25, in recognize
> > rgb_frame = torch.tensor(rgb_frame).cuda().float()
> > File "D:\ProgramData\Anaconda3\envs\Face-Recognition-Access-Control-main\lib\site-packages\torch\cuda__init__.py", line 186, in _lazy_init
> > _check_driver()
> > File "D:\ProgramData\Anaconda3\envs\Face-Recognition-Access-Control-main\lib\site-packages\torch\cuda__init__.py", line 61, in _check_driver
> > raise AssertionError("Torch not compiled with CUDA enabled")
> > AssertionError: Torch not compiled with CUDA enabled
> 
> How should I rewrite my program？

You can rewrite the code in `/Server/ImgServer.py` on lines 25 and 49 as
```python
rgb_frame = torch.tensor(rgb_frame).cuda().float() if facenet.device == "cuda:0" else torch.tensor(rgb_frame).float()
```
and I've submitted a commit <https://github.com/Murlors/Face-Recognition-Access-Control/commit/ed7cebe26612f9ce89f50420966d7c6de2994c9b> to fix this problem.

**Comment by Huyuchao-x** [comment]
> > > > It was addressed in [#2 (comment)](https://github.com/Murlors/Face-Recognition-Access-Control/pull/2#issue-1683324767) , specifically in the commit [GXshuyang@620c395](https://github.com/GXshuyang/Face-Recognition-Access-Control/commit/620c3956cfe9b0bcfd4153c298fbea2d59c43cb1)
> > > 
> > > 
> > > Solved, thank you. But I have another problem: my computer has integrated graphics and I installed pytrch -cpu, but at runtime it happens that：
> > > File "D:/IdeaProject/Face-Recognition-Access-Control-main/Server/ImgServer.py", line 25, in recognize
> > > rgb_frame = torch.tensor(rgb_frame).cuda().float()
> > > File "D:\ProgramData\Anaconda3\envs\Face-Recognition-Access-Control-main\lib\site-packages\torch\cuda__init__.py", line 186, in _lazy_init
> > > _check_driver()
> > > File "D:\ProgramData\Anaconda3\envs\Face-Recognition-Access-Control-main\lib\site-packages\torch\cuda__init__.py", line 61, in _check_driver
> > > raise AssertionError("Torch not compiled with CUDA enabled")
> > > AssertionError: Torch not compiled with CUDA enabled
> > 
> > 
> > How should I rewrite my program？
> 
> You can rewrite the code in `/Server/ImgServer.py` on lines 25 and 49 as
> 
> ```python
> rgb_frame = torch.tensor(rgb_frame).cuda().float() if facenet.device == "cuda:0" else torch.tensor(rgb_frame).float()
> ```
> 
> and I've submitted a commit [ed7cebe](https://github.com/Murlors/Face-Recognition-Access-Control/commit/ed7cebe26612f9ce89f50420966d7c6de2994c9b) to fix this problem.
太感谢你了，你总是精准的解决了我的问题。因为我是第一次接触这方面的程序，我有许多不明白的代码。在我运行ImgServer时，显示如下错误：
  File "D:/IdeaProject/Face-Recognition-Access-Control-main/Server/ImgServer.py", line 35, in recognize
    images = facenet.boxes_to_images(rgb_frame, boxes)
  File "D:\IdeaProject\Face-Recognition-Access-Control-main\Server\FacenetModel.py", line 44, in boxes_to_images
    return_images = [self.loader(image_data.crop(box).resize((160, 160))) for box in boxes]
  File "D:\IdeaProject\Face-Recognition-Access-Control-main\Server\FacenetModel.py", line 44, in <listcomp>
    return_images = [self.loader(image_data.crop(box).resize((160, 160))) for box in boxes]
AttributeError: 'Tensor' object has no attribute 'crop'

**Comment by Murlors** [comment]
> > > > It was addressed in [#2 (comment)](https://github.com/Murlors/Face-Recognition-Access-Control/pull/2#issue-1683324767) , specifically in the commit [GXshuyang@620c395](https://github.com/GXshuyang/Face-Recognition-Access-Control/commit/620c3956cfe9b0bcfd4153c298fbea2d59c43cb1)
> > > > 
> > > Solved, thank you. But I have another problem: my computer has integrated graphics and I installed pytrch -cpu, but at runtime it happens that：
> > > File "D:/IdeaProject/Face-Recognition-Access-Control-main/Server/ImgServer.py", line 25, in recognize
> > > rgb_frame = torch.tensor(rgb_frame).cuda().float()
> > > File "D:\ProgramData\Anaconda3\envs\Face-Recognition-Access-Control-main\lib\site-packages\torch\cuda__init__.py", line 186, in _lazy_init
> > > _check_driver()
> > > File "D:\ProgramData\Anaconda3\envs\Face-Recognition-Access-Control-main\lib\site-packages\torch\cuda__init__.py", line 61, in _check_driver
> > > raise AssertionError("Torch not compiled with CUDA enabled")
> > > AssertionError: Torch not compiled with CUDA enabled
> > > 
> > > How should I rewrite my program？
> > > 
> > You can rewrite the code in `/Server/ImgServer.py` on lines 25 and 49 as
> > ```python
> > rgb_frame = torch.tensor(rgb_frame).cuda().float() if facenet.device == "cuda:0" else torch.tensor(rgb_frame).float()
> > ```
> > and I've submitted a commit [ed7cebe](https://github.com/Murlors/Face-Recognition-Access-Control/commit/ed7cebe26612f9ce89f50420966d7c6de2994c9b) to fix this problem.
> > 
> 太感谢你了，你总是精准的解决了我的问题。因为我是第一次接触这方面的程序，我有许多不明白的代码。在我运行ImgServer时，显示如下错误：
> File "D:/IdeaProject/Face-Recognition-Access-Control-main/Server/ImgServer.py", line 35, in recognize
> images = facenet.boxes_to_images(rgb_frame, boxes)
> File "D:\IdeaProject\Face-Recognition-Access-Control-main\Server\FacenetModel.py", line 44, in boxes_to_images
> return_images = [self.loader(image_data.crop(box).resize((160, 160))) for box in boxes]
> File "D:\IdeaProject\Face-Recognition-Access-Control-main\Server\FacenetModel.py", line 44, in 
> return_images = [self.loader(image_data.crop(box).resize((160, 160))) for box in boxes]
> AttributeError: 'Tensor' object has no attribute 'crop'

I've submitted a commit <https://github.com/Murlors/Face-Recognition-Access-Control/commit/9e7458f05c7ad06a755650ba30c51c3e93122bdd> to try to fix this problem. Can you try the new version? I am currently unable to test it myself since my computer’s camera does not work.
